queries/getRoundDataForUser.ts

```diff
@@ -0,0 +1,36 @@
+import { getSupabaseClient } from "utils/getSupabaseClient";
+
+const supabase = getSupabaseClient();
+
+export const getRoundDataForUser = async (roundId: number, userId: string) => {
+  const { data, error } = await supabase
+    .from("users")
+    .select(
+      `
+  userid,
+  song_selection_votes ( round_id, created_at ),
+  submissions ( round_id, created_at ),
+  sign_ups ( round_id, created_at)
+`
+    )
+    .eq("userid", userId)
+    .eq("song_selection_votes.round_id", roundId)
+    .eq("submissions.round_id", roundId)
+    .eq("sign_ups.round_id", roundId);
+  if (error) {
+    throw new Error(error.message);
+  }
+  if (!data || data.length === 0) {
+    return;
+  }
+
+  const user = data[0];
```

Comment:
this is the smallest nit, but I prefer `const [user] = data`. It's literally the same, but to me it implies that it's all there is in there.